NewForce-Cohort-7/tabloidfullstack-bloodyscallwags
# Trello Ticket #25 - Deactivate User Account

Description
[Trello Ticket #25](https://trello.com/c/4ukkzusr/24-25-deactivate-user-profiles). When viewing the User Profiles List, admin have the ability to deactivate an account.

Testing Instructions
1. git add and git commit whatever you're working on on your branch
2. git checkout main
3. git fetch -a
4. git checkout kw-DeactivateUserProfiles
5. If you don't see the most recent version of my files, git pull origin kw-DeactivateUserProfiles
6. If you haven't created the Tabloid MVC database, please do so.
7. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-7/tabloidfullstack-bloodyscallwags/blob/main/SQL/01_Tabloid_Create_DB.sql)
8. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-7/tabloidfullstack-bloodyscallwags/blob/main/SQL/02_Tabloid_Seed_Data.sql)
9. **Additionally, everyone will need to run this information to create new columns in the database.**

**Run this first**
ALTER TABLE UserProfile
ADD IsActive INT;

**Run this next**
UPDATE UserProfile
SET IsActive = 1;

10. In the SQL query, run SELECT * FROM UserProfile, and you should see a column called IsActive where all values from every current UserProfile is set to 1. 1 will mean the user profile is active.
11. Click the execute with debugger solid green button to launch the app
12. In browser, please go to Login on the nav bar and log in with this email: [foo@bar.com](mailto:foo@bar.com)
13. This should redirect you back to the home page.
14. In the Nav Bar, click User Profiles.
15. Click the Deactivate button on the first account in the row (Aharon Grzeskowski). A pop-up should appear asking you to confirm. Click cancel, and this should take you back to the User List.
16. BUT WAIT! You actually wanted to deactivate Aharon. Click deactivate again, and this time, confirm the deactivation.
17. The page should refresh, and the deactivate button should disappear from Aharon's row, being replaced with "User Account is Deactivated."
18. Now, logout, and then try to login this time with Aharon's email (agrzeskowski3@fc2.comx). You should get a window alert saying the account has been deactivated, and it should not allow you to login.

Checklist:
[] My code follows the style guidelines of this project
[x] I have performed a self-review of my code
[x] I have commented my code, particularly in hard-to-understand areas
[x] My changes generate no new warnings or errors
[x] I have added test instructions that prove my fix is effective or that my feature works
